v3/src/components/graph/components/legend/choroplethLegend/choroplethLegend.ts

```diff
@@ -0,0 +1,112 @@
+// Copyright 2021, Observable Inc.
+// Released under the ISC license.
+// https://observablehq.com/@d3/scale-legend
+import {
+  axisBottom,
+  /*
+  interpolate,
+  interpolateRound,
+  quantize,
+  scaleBand,
+  ScaleContinuousNumeric,
+*/
+  scaleLinear,
+  format,
+  // range,
+  select, range, min, max
+} from "d3"
+import {kChoroplethHeight} from "../../../graphing-types"
+import {neededSigDigitsArrayForQuantiles} from "../../../../../utilities/math-utils"
+
+export type ChoroplethLegendProps = {
+  tickSize?: number,
+  width?: number,
+  rectHeight?: number,
+  transform?: string,
+  marginTop?: number,
+  marginRight?: number,
+  marginLeft?: number,
+  ticks?: number,
+  clickHandler: (quantile: number, extend: boolean) => void,
+  casesInQuantileSelectedHandler: (quantile: number) => boolean
+}
+
+export function choroplethLegend(scale: any, choroplethElt: SVGGElement, props: ChoroplethLegendProps) {
+  const {
+      tickSize = 6, transform = '', width = 320, marginTop = 0, marginRight = 0, marginLeft = 0,
+      ticks = 5, clickHandler, casesInQuantileSelectedHandler
+    } = props,
+    minValue = min(scale.domain()),
+    maxValue = max(scale.domain())
+
+  let tickFormat: any = '.2r',
+    tickValues: any = []
```

Comment:
I think these can be typed as:
```suggestion
  let tickFormat: string | ((i: number) => string) = '.2r',
    tickValues: number[] = []
```